Sources/Public/iOS/LottieAnimationViewBase.swift

```diff
@@ -39,7 +39,11 @@ open class LottieAnimationViewBase: UIView {
 
   var screenScale: CGFloat {
     #if os(iOS) || os(tvOS)
-    UIScreen.main.scale
+    if #available(iOS 13.0, *) {
+        return window?.windowScene?.screen.scale ?? .zero
```

Comment:
Is there a way to provide a reasonable fallback, similar to `UIScreen.main.scale`, rather than just `.zero`? Returning `.zero` for some reason would result in a very broken experience.

Reply:
`UIScreen.main.scale` has several values taht each device display.

>  For Retina displays, the scale factor may be 3.0 or 2.0 and one point can represented by nine or four pixels, respectively. For standard-resolution displays, the scale factor is 1.0 and one point equals one pixel
> https://developer.apple.com/documentation/uikit/uiscreen/1617836-scale

Not all devices had a Retina display for the iPhone. This is expected complex control flow.

If there noting special happens, a view will have `window.windowScene`. However, it seems reasonable to set it to `1.0` to ensure minimal operation through the scale factor.

Reply:
- Related commit history : be10a85